### PR TITLE
Courses

### PR DESCRIPTION
**Pull request by lorenzo-sani**
@willu47 I just updated the way courses are created using a front matter variable "course", writing it to the lectures and recalling it in the course page

### CODE REVIEW

**Comment by willu47** [comment]
Hi @lorenzo-sani - looks like a good addition.  There's a lot of conflicts here though. You need to follow these steps after a pull request happens:
```bash
git checkout gh-pages # change back to the gh-pages branch
git pull # pull down the latest changes
git checkout -b lorenzo/courses. # create and switch to a new branch

# Now do some work
git add <do some work>
git commit -m "Commit message"

# Push up the changes and submit a pull request
git push -u origin lorenzo/courses
```

This way, you ensure that your local copy of the repository never falls too far out of sync with the one on Github.

**Comment by willu47** [comment]
You can also delete you old brach too:

`git branch -d old_branch_name`

The small `-d` is a "safe" delete - it will only remove the branch if Git knows that the commits on that branch have been merged in to the `gh-pages` (normally `master` branch)

**Comment by lorenzo-sani** [comment]
@willu47  thanks for the clarification and sorry for the mess 😅

**Comment by willu47** [comment]
Hi Lorenzo - I see a number of duplicate lectures now - can you remove these?

Also, I'm no convinced of this automated route. Currently this is a middle ground between automatically generated course pages (generated from `course:` tags in the lectures) and fully manual courses - where a collection of course pages contain links to the lectures, and tags are not used.

The fully automatic route raises the issue of the order of the lectures. How do you define which lecture goes first, second, third etc. What if you wish to include a lecture in multiple courses?

In other words, I think the automatic generation is cool, but it raises more questions than it answers.  I suggest we go for a manual approach.  Perhaps add the functionality as an issue to keep a record (together with the code snippets) and we can revisit this at another point in time.

**Comment by lorenzo-sani** [comment]
Actually, using  `{% if lecture.course contains page.title%}` allows for a lecture to be added to more than one course.
But I agree with the problem of deciding the order of the lectures, I hadn't thought it through. 
I agree with removing that part for now and stick for the manual approach. However, I would maintain the `course` category since it could be useful in some other instances (for example creating the navigation bar).
